packages/s2-core/src/interaction/base-interaction/click/data-cell-click.ts

```diff
@@ -59,16 +60,20 @@ export class DataCellClick extends BaseEvent implements BaseEventImplement {
         return;
       }
 
+      const headerSelectedNode = getRowColHeaderCellBySelectedCellHighlight(
+        meta.id,
+        this.spreadsheet,
+      );
+
       interaction.changeState({
```

Comment:
这里没必要去修改 changeState 这个方法，你可以参考  `col-brush-selection.ts` 和 `row-brush-selection.ts`  中 `updateSelectedCells` 的更新思路，将更新的单元格范围自己控制。
```suggestion
      interaction.changeState({
      cells: map(this.brushRangeCells, getCellMeta),
      stateName: InteractionStateName.SELECTED,
      onUpdateCells: (root) => {
        root.updateCells(root.getAllColHeaderCells());
      },
    });
```